---
title: "Kallisto Alignment Yamagishi Reads"
author: "Katalina Bobowik"
date: "8/15/2017"
output: pdf_document
---

First, download human cDNA from ENSEMBL and transcript index
```{bash}
# over ftp; ENSEMBL release 89
wget ftp://ftp.ensembl.org/pub/current_fasta/homo_sapiens/cdna/Homo_sapiens.GRCh38.cdna.all.fa.gz
```


Next , build an index from HG38 cDNA transcripts
```{bash}
kallisto index -i ~/Kallisto/HG38_transcripts.idx /vlsci/SG0008/shared/genomes/hg38/Homo_sapiens.GRCh38.cdna.all.fa.gz
```

Quantification for single-end reads. Note: the fragment lenth is unknown, so as per the google groups site (https://groups.google.com/forum/#!topic/kallisto-sleuth-users/h5LeAlWS33w), 200bp is used. For more accuracy, and still with the sake of time considered, 100 bootstraps are used (parameter "-b").
```{bash}

# Create text file for array- one command for each file.
for i in `cat /scratch/SG0008/kbobowik/SRR_Acc_List.txt`; do
  echo kallisto quant -i ~/Kallisto/HG38_transcripts.idx -t 4 -o ~/Kallisto/sample_output/Hg38/${i} -b 100 --single -l 200 -s 20 /scratch/SG0008/kbobowik/SRA_Files/${i}.fastq.gz
done > Kallisto_ArrayTable.txt
```

Execute with an array submission script (sbatch kallisto.sh)
```{bash}

# Array submission script saved as kallisto.sh, saved in kbobowik/scripts folder

#!/bin/bash
# Created by the Melbourne Bioinformatics job script generator for SLURM
# Mon Aug 15 2017 13:14:21 GMT+1000 (AEST)

# Array set up:
#SBATCH --array=1-122

# Partition for the job:
#SBATCH -p sysgen

# The name of the job:
#SBATCH --job-name="Kallisto"

# The project ID which this job should run under:
#SBATCH --account="SG0008"

# Maximum number of tasks/CPU cores used by the job:
#SBATCH --ntasks=1
#SBATCH --mem-per-cpu=2048

# Send yourself an email when the job:
# aborts abnormally (fails)
#SBATCH --mail-type=FAIL
# begins
#SBATCH --mail-type=BEGIN
# ends successfully
#SBATCH --mail-type=END

# Use this email address:
#SBATCH --mail-user=katalina.bobowik@gmail.com

# Export path variables.
#SBATCH --export=ALL

# The maximum running time of the job in days-hours:mins:sec
#SBATCH --time=2:00:00

# Output control:
#SBATCH --workdir="/scratch/SG0008/kbobowik/logs/"
#SBATCH --output="Kallisto_%a.log"

# check that the script is launched with sbatch
if [ "x$SLURM_JOB_ID" == "x" ]; then
   echo "You need to submit your job to the queuing system with sbatch"
   exit 1
fi

# Run the job from the directory where it was launched (default)

# Run the simulations:
commandToRun=`head -n ${SLURM_ARRAY_TASK_ID} /vlsci/SG0008/kbobowik/Kallisto/Kallisto_ArrayTable.txt | tail -n 1`
eval $commandToRun
```

Lastly, tidy up files by removing decimals
```{bash}

# Remove decimals from ensembl IDs
for file in `ls -d /home/users/ntu/kbobowik/scratch/Kallisto/Human_PF_Combined/*/`; do
echo $(basename $file)
awk -F "\t" '{gsub(/\..*$/,"",$1)}1' OFS="\t" ${file}/abundance.tsv > ${file}/noDecimal_abundance.tsv
done
```
